### PR TITLE
Add convenience options to 'add_cmdlet' and 'add_script'

### PR DESCRIPTION
**Pull request by malthe**

### CODE REVIEW

**Comment by malthe** [comment]
@jborean93 not sure this is really a good idea. My use-case is that I wanted users to be able to easily have an "init script" such as:
```powershell
> Set-PSDebug -Trace 1
```
And it seemed a little awkward with the composable interface.

**Comment by jborean93** [comment]
I'm unsure how I feel about this one. The current setup is designed to replicate the [PowerShell class](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.powershell?view=powershellsdk-7.0.0) to the best it can. The `add_cmdlet` is already a convenience operation as you can't normally do overloading in a simple way that `AddCommand` offers and the latter is reserved for when adding an actual `Command` instance rather than just the string. This is certainly easier to do compared to before but even so I'm not 100% convinced before was even that hard to begin with.

```python
ps = PowerShell(rp)
ps.add_cmdlet('My-Function').add_parameters({...}).add_statement()

# vs

ps.add_cmdlet('My-Function', parameters={...}, end_of_statement=True)
```

In fact `end_of_statement` makes it longer than the original so you don't really save on typing there. I'm happy to be convinced otherwise but I'm somewhat leaning towards just keep it the same as the .NET API.

File: pypsrp/powershell.py [hunk 1]
```diff
@@ -890,19 +902,22 @@ def add_parameters(self, parameters):
             self.add_parameter(parameter_name, value)
         return self
 
-    def add_script(self, script, use_local_scope=None):
+    def add_script(self, script, use_local_scope=None, end_of_statement=False):
```
Comment (jborean93) [inline pypsrp/powershell.py:905]: Just as an FYI a script can also have parameters.

**Comment by malthe** [comment]
I suppose I could instead prepare a "canned" command, which could easily be imported and added to the pipeline.
```python
debug = Command("Set-PSDebug", args=CommandParameter("Trace", 1), end_of_statement=True)
```
I'll go ahead and close out this pull request.

**Pull request closed without merging**